### PR TITLE
Fix syn

### PR DESCRIPTION
**Pull request by Frky**
Fix bug in **masscanned**.

**Current behaviour:** **masscanned** answers to any packet with the `SYN` flag, no matter what other flag is present.

**Expected behaviour:** **masscanned** should only answer `SYN` packets (with no other flag).

### CODE REVIEW

File: src/layer_4/tcp.rs [hunk 1]
```diff
@@ -105,7 +105,7 @@ pub fn repl<'a, 'b>(
             tcp_repl.set_sequence(tcp_req.get_acknowledgement());
         }
         /* Answer to SYN */
-        flags if flags & TcpFlags::SYN == TcpFlags::SYN => {
+        flags if flags == TcpFlags::SYN => {
```
Comment (p-l-) [inline src/layer_4/tcp.rs:108]: I'm not sure to agree here: it is possible to have other flags set (SYN+ECN+CWR for example).

In firewall rules we often specify `flags S/SA` which means "from syn and ack flags, only syn is set". Maybe we could do the same here?

```suggestion
        flags if flags & (TcpFlags::SYN | TcpFlags::ACK) == TcpFlags::SYN => {
```

Comment (Frky) [reply src/layer_4/tcp.rs:108]: What about packets with other flags that would be inconsistent, as for example `SYN` + `FIN`?

Comment (p-l-) [reply src/layer_4/tcp.rs:108]: Good point. No idea. Should we mimic "classical" networking stacks?

**Comment by Frky** [comment]
Ok, I did some tests on a Linux network stack (Debian 12).

Script:
```
#!/usr/bin/env python3

from scapy.all import *

res = open("res.txt", "w")
for f in range(0, 256):
    p = IP(dst="10.99.99.0")/TCP(sport=55555, dport=8080, flags=f)
    a = sr1(p, iface="vethtest", timeout=0.1)
    if a:
        res.write(f"{p[TCP].flags} > {a[TCP].flags}\n")
res.close()
```

Result:
```
S > SA
SP > SA
A > R
FA > R
SA > R
FSA > R
PA > R
FPA > R
SPA > R
FSPA > R
SU > SA
SPU > SA
AU > R
FAU > R
SAU > R
FSAU > R
PAU > R
FPAU > R
SPAU > R
FSPAU > R
SE > SA
SPE > SA
AE > R
FAE > R
SAE > R
FSAE > R
PAE > R
FPAE > R
SPAE > R
FSPAE > R
SUE > SA
SPUE > SA
AUE > R
FAUE > R
SAUE > R
FSAUE > R
PAUE > R
FPAUE > R
SPAUE > R
FSPAUE > R
SC > SA
SPC > SA
AC > R
FAC > R
SAC > R
FSAC > R
PAC > R
FPAC > R
SPAC > R
FSPAC > R
SUC > SA
SPUC > SA
AUC > R
FAUC > R
SAUC > R
FSAUC > R
PAUC > R
FPAUC > R
SPAUC > R
FSPAUC > R
SEC > SAE
SPEC > SAE
AEC > R
FAEC > R
SAEC > R
FSAEC > R
PAEC > R
FPAEC > R
SPAEC > R
FSPAEC > R
SUEC > SAE
SPUEC > SAE
AUEC > R
FAUEC > R
SAUEC > R
FSAUEC > R
PAUEC > R
FPAUEC > R
SPAUEC > R
FSPAUEC > R
```

If we assume we don't want to be sending `RST` packets, **masscanned** should answer to:
```
S
SP
SU
SPU
SE
SPE
SUE
SPUE
SC
SPC
SUC
SPUC
```

which is:
* `S` + any combination of `P|U|E`
* or `S` + any combination of `P|U|C`.

In other words, `S` + any combination of `P|U|C|E` + not (`E` **and** `U`).

I will fix my PR.

**Comment by p-l-** [comment]
Thanks for taking the time to run this comprehensive test @Frky!

**Review by p-l- — APPROVED** [review]

**Pull request merged**